File: Codeforces/red_and_blue_beans.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;



int CheckValid(int x[]){
    int possible = 1;
    int min_among = std::min(x[0], x[1]);

    int main_array[min_among][2];
    for (int i; i<min_among; i++){
        main_array.push_back([1, 1]);
    }

    int difference_between = std::abs(x[0]- x[1]);
    while (difference_between>0){
        for (int j; j<min_among; j++){
            if (std::abs(main_array[j][0] - main_array[j][1] < x[2])){
                main_array[j][1] += 1;
                difference_between -= 1;
            }
            else{
                possible = 0;
                difference_between = 0;
                break;
            }
        }
    }
    return possible;
    
}

int main(){
    int n;
    cin>> n;
    for (int i; i<n; i++){
        int a, b, c;
        cin >> a >> b >> c;
        int somearray[] = {a, b, c};
        CheckValid(somearray);        
    }

}

File: Level-3 Codeforces/Can-1/red_and_blue_beans.py



def CheckValid(x):
    possible = 'YES'
    minimum_among = min(x[0], x[1])

    main_array = []
    for i in range(minimum_among):
        main_array.append([1,1])

    # print(main_array)

    difference_between = abs(x[0]-x[1])
    while difference_between>0:
        for i in range(minimum_among):
            # if difference_between > 0:
            if abs(main_array[i][0] - main_array[i][1]) < x[2]:
                main_array[i][1] += 1
                difference_between -= 1
            else:
                possible = "NO"
                difference_between = 0
                break
        

            # print(main_array)

    return possible






if __name__ == '__main__':
    n = int(input())
    for ii in range(n):
        x = list(map(int, input().split()))
        print(CheckValid(x))
        